### PR TITLE
Change all template paths to be project-scoped

### PR DESCRIPTION
**Pull request by abelanger5**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [X] Other (please describe): improve scoping of template endpoints

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

Endpoints to retrieve templates are user-scoped, rather than project-scoped. 

## What is the new behavior?

Make all template endpoints project-scoped. We've moved the project-scoped endpoints to `/api/v1` and we'll keep the old user-scoped template endpoints around for a few releases to prevent immediate breaking changes.

### CODE REVIEW

File: api/client/template.go [hunk 1]
```diff
@@ -26,14 +27,16 @@ func (c *Client) ListTemplates(
...
+	projectID uint,
 	name, version string,
 	req *types.GetTemplateRequest,
 ) (*types.GetTemplateResponse, error) {
 	resp := &types.GetTemplateResponse{}
 
 	err := c.getRequest(
 		fmt.Sprintf(
-			"/templates/%s/%s",
+			"/v1/projects/%d/templates/%s/%s",
```
Comment (mnafees) [inline api/client/template.go:38]: Can we keep this consistent with the `GET /api/v1/projects/{project_id}/clusters/{cluster_id}/namespaces/{namespace}/env_groups/{name}/versions/{version}` v1 API endpoint and scope it with the `/versions/` prefix?

File: api/server/handlers/v1/template/get.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	if !isValid {
+		t.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(
+			fmt.Errorf("invalid repo_url parameter"),
+			http.StatusBadRequest,
+		))
+
+		return
+	}
+
+	name, _ := requestutils.GetURLParamString(r, types.URLParamTemplateName)
```
Comment (mnafees) [inline api/server/handlers/v1/template/get.go:63]: Should we be checking for the existence of this URL param? If it is empty the `LoadChartPublic()` might throw an error, right?

File: api/server/handlers/v1/template/get.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		return
+	}
+
+	parserDef := &parser.ClientConfigDefault{
+		HelmChart: chart,
+	}
+
+	res := &types.GetTemplateResponse{}
+	res.Metadata = chart.Metadata
+	res.Values = chart.Values
```
Comment (mnafees) [inline api/server/handlers/v1/template/get.go:88]: We should set the `RepoURL` field as well here for more transparency.

File: api/server/handlers/v1/template/get_upgrade_notes.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	if !isValid {
+		t.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(
+			fmt.Errorf("invalid repo_url parameter"),
+			http.StatusBadRequest,
+		))
+
+		return
+	}
+
+	name, _ := requestutils.GetURLParamString(r, types.URLParamTemplateName)
```
Comment (mnafees) [inline api/server/handlers/v1/template/get_upgrade_notes.go:63]: Same validation as the above comment.

File: api/server/router/v1/project.go [hunk 5]
```diff
@@ -61,5 +86,185 @@ func getV1ProjectRoutes(
...
+	//   '403':
+	//     description: Forbidden
+	getTemplateEndpoint := factory.NewAPIEndpoint(
+		&types.APIRequestMetadata{
+			Verb:   types.APIVerbGet,
+			Method: types.HTTPVerbGet,
+			Path: &types.Path{
+				Parent: basePath,
+				RelativePath: fmt.Sprintf(
+					"%s/templates/{%s}/{%s}",
```
Comment (mnafees) [inline api/server/router/v1/project.go:179]: Same as above comment in `api/client/template.go` about the `/versions/` prefix.

**Review by mnafees — CHANGES_REQUESTED** [review]
Just a few comments, but looks great overall 👍🏽

**Review by mnafees — APPROVED** [review]

**Review by meehawk — APPROVED** [review]

**Pull request merged**